src/Microsoft.OpenApi/Services/OpenApiWalker.cs

```diff
@@ -449,11 +573,56 @@ internal void Walk(OpenApiRequestBody requestBody)
             Walk(requestBody as IOpenApiExtensible);
         }
 
+        /// <summary>
+        /// Visits dictionary of <see cref="OpenApiHeader"/>
+        /// </summary>
+        internal void Walk(IDictionary<string, OpenApiHeader> headers)
+        {
+            if (headers == null)
+            {
+                return;
+            }
+
+            _visitor.Visit(headers);
+            if (headers != null)
+            {
+                foreach (var header in headers)
+                {
+                    Walk(header.Key.Replace("/", "~1"), () => Walk(header.Value));
```

Comment:
>.Replace("/", "~1") [](start = 35, length = 19)

nit, I thought we already handled this somewhere else. If not, can we do this in the VisitorBase.Enter, given that this is a JSON pointer requirement.